content-retrievers/langchain4j-community-lucene/src/main/java/dev/langchain4j/rag/content/retriever/lucene/DirectoryFactory.java

```diff
@@ -0,0 +1,51 @@
+package dev.langchain4j.rag.content.retriever.lucene;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.MMapDirectory;
+
+/**
+ * Static factory for create Lucene directories.
+ */
+public class DirectoryFactory {
+
+    /**
+     * Create a memory mapped file-system based directory.
+     *
+     * @param directoryPath Path for the directory.
+     *
+     * @return Lucene directory
+     */
+    public static Directory fsDirectory(final Path directoryPath) {
```

Comment:
```suggestion
    public static Directory fsDirectory(Path directoryPath) {
```

Coult you please remove all `final` in method parameters and local variables?